### PR TITLE
Stories/us1140 get pregnancy data from cprs

### PR DESCRIPTION
**Pull request by alinafrey**
Update Lactation and Pregnancy data in Maternity Tracker with the data from CPRS

### CODE REVIEW

File: Dashboard/va.gov.artemis.commands/Dsio/Pregnancy/WvrpcorGetPregDetailsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    var lmpDate = Util.Piece(this.Response.Lines[0], "LMP: ", 2);
+                    var eddDate = Util.Piece(this.Response.Lines[1], "EDD: ", 2);
+                    this.EnteredBy = Util.Piece(this.Response.Lines[3], "Entered by: ", 2);
+                    enteredOn = Util.Piece(this.Response.Lines[4], "Entered on: ", 2);
+                    this.EDD = VistaDates.StandardizeDateFormat(eddDate);
+                    this.LMP = VistaDates.StandardizeDateFormat(lmpDate);
+                }
+                else
+                {
+                    this.EnteredBy = Util.Piece(this.Response.Lines[1], "Entered by: ", 2);
```
Comment (aujosh84) [inline Dashboard/va.gov.artemis.commands/Dsio/Pregnancy/WvrpcorGetPregDetailsCommand.cs]: What if the patient has never been pregnant? will these values still be there?

File: Dashboard/va.gov.artemis.ui.data/Brokers/Pregnancy/PregnancyRepository.cs [hunk 2]
```diff
@@ -220,12 +220,203 @@ public PregnancyResult GetCurrentPregnancy(string patientDfn)
...
+            var secondArg = "1";
+            getWvrpcorWomensHealthDataCommand.AddCommandArguments(patientDfn, secondArg);
+            RpcResponse wvrpcorResponse = getWvrpcorWomensHealthDataCommand.Execute();
+            result.SetResult(wvrpcorResponse.Status == RpcResponseStatus.Success, wvrpcorResponse.InformationalMessage);
+
+            if (result.Success)
+            {
+                currentWvrpcorPregnancy.PatientDfn = getWvrpcorWomensHealthDataCommand.PatientDfn;
+                currentWvrpcorPregnancy.RecordType = PregnancyRecordType.Current;
+                //currentWvrpcorPregnancy.Ien = getWvrpcorWomensHealthDataCommand.PregnancyIen;
```
Comment (aujosh84) [inline Dashboard/va.gov.artemis.ui.data/Brokers/Pregnancy/PregnancyRepository.cs]: dead code.

File: Dashboard/va.gov.artemis.ui.data/Brokers/Pregnancy/PregnancyRepository.cs [hunk 3]
```diff
@@ -220,12 +220,203 @@ public PregnancyResult GetCurrentPregnancy(string patientDfn)
...
+                                DateTime eddDsio = currentPregnancyDsio.EDD;
+                                DateTime eddWvrpcor = currentPregnancyWvrpcor.EDD;
+                                string lmpDsio = currentPregnancyDsio.Lmp;
+                                string lmpWvrpcor = currentPregnancyWvrpcor.Lmp;
+
+                                //If the current pregnancy in Maternity Tracker is different than the one in CPRS,
+                                //update it with the pregnancy data from CPRS     
+                                if (eddDsio != eddWvrpcor || lmpDsio != lmpWvrpcor)
+                                {
+                                    //currentPregnancyDsio.Ien = ienWvrpcor;
```
Comment (aujosh84) [inline Dashboard/va.gov.artemis.ui.data/Brokers/Pregnancy/PregnancyRepository.cs]: dead code

File: Dashboard/va.gov.artemis.commands/Dsio/Pregnancy/WvrpcorGetPregDetailsCommand.cs (hunk 1 above)
Comment (alinafrey) [reply Dashboard/va.gov.artemis.commands/Dsio/Pregnancy/WvrpcorGetPregDetailsCommand.cs]: Excellent point! Great catch!!

**Review by aujosh84 — APPROVED** [review]

**Pull request merged**